_posts/2020-12-16-solidity-v0.8.0-release-announcement.md

```diff
@@ -0,0 +1,240 @@
+than there are positive numbers. This means that the following code will trigger a ``Panic``:
+
+```solidity
+function f() pure {
+    int8 x = -128;
+    -x;
+}
+```
+
+The reason is that ``128`` does not fit an 8-bit signed integer.
+
+For the same reason, signed division can result in an assertion:
+```solidity
+function f() pure {
+    int8 x = -128;
+    x/(-1);
+}
+```
+
+Everyone who wanted to write a ``SafeMath`` function for exponentiation probably noticed
+that it is rather expensive to do that because the EVM does not provide overflow signalling.
+Essentially you have to implement your own ``exp`` routine without using the ``exp`` opcode
+for the general case.
+
+We hope that we found a rather efficient implementation and would also appreciate your feedback about that!
+
+For many special cases, we actually implemented it using the ``exp`` opcode instead of our own implementation.
+More specifically, exponentiation operations that use a literal number as base will use the ``exp`` opcode directly.
+There are also specialized code paths for bases that are variables with small value. For these,
+the exponent is compared to a hard-coded safe upper bound. These bases (or base ranges) are:
+The special bases 0, 1 and 2, all bases between 3 and 10 and the bases between 11 and 306.
+If the base lies in one of these ranges and the exponent is small enough, it
+should result in a cheap ``exp`` opcode.
+
+#### ``unchecked`` Blocks
+
+Since checked arithmetic uses more gas, as discussed in the previous section, and
+because there are times when you really want the wrapping behaviour, for example when
+implementing cryptographic routines, we provide a way to disable the checked arithmetic and
+re-enable the previous "wrapping" or "modulo" arithmetic:
+
+Any block of the form ``unchecked { ... }`` uses wrapping arithmetic. You can use
+this special block as a regular statement inside another block:
+
+```solidity
+contract C {
+    function f() public pure returns (uint) {
+        uint x = 0;
+        unchecked { x--; }
+        return x;
+    }
+}
+```
+
+An unchecked block can of course contain any number or any kind of statement
+and will create its own variable scope.
+If there is a function call inside the unchecked block, the function will
+not inherit the setting - you have to use another ``unchecked`` block inside the function
+if you also want it to use wrapping arithmetic.
+
+We decided for some restrictions for the ``unchecked`` block:
+
+1. In modifiers you cannot use ``_;`` inside it, because this might lead to confusion about whether the property is inherited.
+
+2. You can only use it inside a block but not as a replacement for a block. The following code snippets, for example, are both invalid:
+
+    ```solidity
+    function f() unchecked { uint x = 7; }
+    ```
+
+    ```solidity
+    function f() pure {
+        uint x;
+        for (uint i = 0; i < 100; i++) unchecked { x += i; }
```

Comment:
```suggestion
    ```solidity
    function f() pure {
        uint x;
        for (uint i = 0; i < 100; i++) unchecked { x += i; }
    }
```
Added a missing brace